### PR TITLE
NLS Globalisation Feature : Requirement for ArtEST Project

### PR DESCRIPTION
**Pull request by Vyshantha**
## Why is this PR needed
The NLS or i18n feature is being added as part of the requirement raised in the ArtEST project at UoC/UzK. 

## What purpose does this PR serve
This feature would enable switching between these languages ( English [en], German [de], Mongolian in Cyrillic alphabet [mn], Mongolian traditional script [mon], Greek [el] and Italian [it] ) - once the required strings are translated. The Kompakkt Viewer interface is refreshed through the URL query parameter ( e.g. `https://<Kompakkt_URL>/?locale=mn` ) and the UI would be shown in the Mongolian Cyrillic alphabet.

## Which updates are expected
The changes involved are in the HTML elements like `placeholder`, `MatTooltip`, `attr.aria-label`, `label`, `<mat-*>` UI components, `HTML tags` with text based on this template which is served by the Translation Service that is included in the app.component and app.module TS files
> '<String_to_be_translated>' | translate

### CODE REVIEW

**Comment by HeyItsBATMAN** [comment]
@lozanaross @ZetOE

Our new colleague @Vyshantha has put in necessary work for internationalization of the Viewer and the Repo. Now, for our project partners of the ARTEST project to test if the Mongolian translations are working correctly, we will have to merge this pretty soon (in the next 2 weeks).

If you can take a look at the changes? Just so you are updated on what's happening to Kompakkt in the context of ARTEST.

TL;DR of this PR is that a new `TranslationService` and a `TranslatePipe` has been introduced. A user that has their computer set to a Mongolian locale or appends the `?locale=mn` query to the URL will be shown a partly translated version of Kompakkt.

As i18n affects the whole application, this PR changes a lot of files.
This also affects the Repo here: https://github.com/Kompakkt/Repo/pull/29

**Comment by lozanaross** [comment]
Thanks for the heads up @HeyItsBATMAN - I think it's good to also tag @luguenth in such issues, so he's in the loop. I am not sure if we'll implement multilinguality yet, so it might also be good to discuss how to best handle distinct branches of the product in the future.
